src/transactions/feed/detailContent/JumpstartContent.tsx

```diff
@@ -1,35 +1,168 @@
+    abi: walletJumpstart.abi,
+    functionName: 'erc20Claims',
+    args: [beneficiary, index],
+  })
+
+  // @ts-ignore
+  const claimed: boolean = erc20Claim[3]
+
+  Logger.debug(`Reward deposited in ${transactionHash} ${claimed ? 'was' : 'was NOT'} claimed`)
+
+  return { beneficiary, index: Number(index), claimed }
+}
+
+function JumpstartContent({ transfer, showOnBottomSheet }: Props) {
   const { t } = useTranslation()
-  const transferTokenInfo = useTokenInfo(transfer.amount.tokenId)
 
+  const transferTokenInfo = useTokenInfo(transfer.amount.tokenId)
   const parsedAmount = new BigNumber(transfer.amount.value).abs()
   const token = useTokenInfo(transfer.amount.tokenId)
   const isDeposit = transfer.type === TokenTransactionTypeV2.Sent
 
+  const walletAddress = useSelector(walletAddressSelector)
+  const jumpstartContractAddress = getJumpstartContractAddress(transfer.networkId)
+
+  const fetchEscrowData = useAsync(
+    async () => {
+      if (!isDeposit) {
+        return
+      }
+
+      return await getRewardDataAndStatus(transfer.networkId, transfer.transactionHash as Hash)
+    },
+    [],
+    {
+      onError: (error) => {
+        // TODO: Show error message in the UI
+        Logger.error(TAG, 'Failed to fetch escrow data', error)
+      },
+    }
+  )
+
   if (!token) {
     // should never happen
     Logger.error(TAG, 'Token is undefined')
     return null
   }
 
-  // TODO: Add reclaim button
+  const ClaimedLabel = () => (
+    <View style={[styles.claimedContainer]}>
+      <>
+        <Text style={[styles.claimedText]}>{'Claimed'}</Text>
+        <Checkmark color={Colors.successDark} width={Spacing.Thick24} height={Spacing.Thick24} />
+      </>
+    </View>
+  )
+
+  const onReclaimPress = () => {
+    if (!fetchEscrowData.result) {
+      Logger.error(TAG, 'No escrow data found when trying to reclaim')
+      return
+    }
+
+    const { beneficiary, index } = fetchEscrowData.result
+
+    const reclaimTx: SerializableTransactionRequest = {
+      from: walletAddress as Address,
+      to: jumpstartContractAddress as Address,
+      value: '0',
+      data: encodeFunctionData({
+        abi: walletJumpstart.abi,
+        functionName: 'reclaimERC20',
+        args: [beneficiary, index],
+      }),
+      // TODO: Estimate
+      gas: '130000',
+      maxFeePerGas: '1000000000',
+    }
+
+    navigate(Screens.JumpstartReclaimBottomSheet, {
+      reclaimTx,
+      networkId: transfer.networkId,
+      tokenAmount: {
+        value: parsedAmount.toString(),
+        tokenId: token.tokenId,
+        tokenAddress: token.address ?? undefined,
+        localAmount: transfer.amount.localAmount,
+      },
+    })
+  }
+
+  const ReclaimButton = () => (
+    <View style={styles.buttonContainer}>
+      {fetchEscrowData.result?.claimed ? (
+        <ClaimedLabel />
+      ) : (
+        <Button
+          showLoading={fetchEscrowData.loading}
+          disabled={fetchEscrowData.loading}
```

Comment:
we need a separate loading button right? otherwise when loading whether a user can reclaim, the button will show "Claim" + loader?